core/autoscaling.go

```diff
@@ -526,12 +573,27 @@ func (a *autoScalingGroup) launchCheapestSpotInstance(
 	}
 	logger.Println("Found on-demand instance", *baseInstance.InstanceId)
 
-	newInstanceType, err := baseInstance.getCheapestCompatibleSpotInstanceType()
-
+	newInstanceType, err = a.getInstanceTypeByTagInASG()
 	if err != nil {
-		logger.Println("No cheaper compatible instance type was found, "+
-			"nothing to do here...", err)
-		return errors.New("no cheaper spot instance found")
+		logger.Println("Errors while trying to fetch tags for instance type...")
+		return errors.New("Errors while trying to fetch tags")
+	}
+
+	// Check option to keep instance type
+	// If we keep the instance type we don't need to calculate the
+	// compatible instance type.
+	// Also ignore if it's already specified in the instance-type tag
+	if newInstanceType == "" {
+		if a.region.conf.KeepInstanceType {
+			newInstanceType = *baseInstance.InstanceType
+		} else {
+			newInstanceType, err = baseInstance.getCheapestCompatibleSpotInstanceType()
+		}
```

Comment:
I think that having the option as a `string` rather than a `bool` would also potentially simplify this `if` / `else` as the type would be already known completely or not at all, thus removing the possibility of a temporary `""` type.